lib/DoorControl/Controller/DoorControl.pm

```diff
@@ -24,17 +26,20 @@ sub unlock {
     
     my $response->{result} = 0;
     
-    if($results->{authorized} == 1) {
+    if($results->{authorized} == 1 || $self->internal || ($results->{authorized} == 2 && Date_IsWorkDay(ParseDate('now'), 1))) {
         $response->{result} = 1;
```

Comment:
It appears that there's a lot of duplicate code as Command::getty.pm.  Duplicate code should get moved into a model.  Maintaining consistency as the program changes becomes very difficult, and downright impossible.  For example, in the future you may be asked to add a feature or fix a bug specifically in the Web-based version.  Not thinking, you'll update this file but forget to "sync" the changes and not apply the same updates to the getty.pm.